File: ruby/classes/class_components.rb
# Copyright (C) 2019-2026 Katarzyna Krasińska
# Languagium - https://github.com/katheroine/languagium
# Licensed under GPL-3.0 - see LICENSE.md

class Account
  attr_accessor :login, :email, :is_active, :id, :connections_number

  CONNECTIONS_MAX_NUMBER = 10

  def display()
    print("ID: ", @id, "\n",
      "Login: ", @login, "\n",
      "E-mail: ", @email, "\n",
      "Is active: ", @is_active , "\n",
      "Has connections: ", has_connections(), "\n"
    )
  end

  def has_connections()
    return @connections_number ? (@connections_number > 0) : false
  end
end

timothy = Account.new

timothy.login = "tim"
timothy.email = "timothy.muppetone@gmail.com"
timothy.is_active = true

timothy.display()

File: python/classes/class_components.py
# Copyright (C) 2019-2026 Katarzyna Krasińska
# Languagium - https://github.com/katheroine/languagium
# Licensed under GPL-3.0 - see LICENSE.md
# SPDX-License-Identifier: GPL-3.0-or-later

class Account:
    login = ""
    email = ""
    is_active = False
    id = 0

    connections = []
    connections_number = 0

    def display(self):
        print("ID:", self.id,
          "\nLogin:", self.login,
          "\nE-mail:", self.email,
          "\nIs active:", self.is_active,
          "\nHas connections:", self.has_connections()
        );

    def has_connections(self):
        return (self.connections_number > 0)

timothy = Account()

timothy.login = "tim"
timothy.email = "timothy.muppetone@gmail.com"
timothy.is_active = True

timothy.display()
